cmd/bundle/main.go

```diff
@@ -154,3 +178,163 @@ func resolveAuthBasedOnTarget(ref name.Reference) (authn.Authenticator, error) {
 	}), nil
 }
+
+// Prune removes the image from the container registry
+//
+// Deleting a tag, or a whole repo is not as straightforward as initially
+// planned as DockerHub seems to restrict deleting a single tag for
+// standard users. This might be subject to change, but as of September
+// 2021 it is limited to the business tier. However, there is an API call
+// to delete the whole repository. In case there is only one tag used in
+// a repository, the effect is pretty much the same. For convenience, there
+// is a provider switch to deal with images on DockerHub differently.
+//
+// DockerHub images:
+// - In case the repository only has one tag, the repository is deleted.
+// - If there are multiple tags, the tag to be deleted is overwritten
+//   with an empty image (to remove the content, and save quota).
+// - Edge case would be no tags in the repository, which is ignored.
+//
+// Other registries:
+// Use standard spec delete API request to delete the provided tag.
+//
+func Prune(ctx context.Context, image string, auth authn.Authenticator) error {
+	ref, err := name.ParseReference(image)
+	if err != nil {
+		return err
+	}
+
+	switch ref.Context().RegistryStr() {
+	case "index.docker.io":
+		list, err := remote.List(ref.Context(), remote.WithContext(ctx), remote.WithAuth(auth))
+		if err != nil {
+			return err
+		}
+
+		switch len(list) {
+		case 0:
+			return nil
+
+		case 1:
+			var authr *authn.AuthConfig
+			authr, err = auth.Authorization()
+			if err != nil {
+				return err
+			}
+
+			var token string
+			token, err = dockerHubLogin(authr.Username, authr.Password)
+			if err != nil {
+				return err
+			}
+
+			return dockerHubRepoDelete(token, ref)
+
+		default:
+			log.Printf("Removing a specific image tag is not supported on %s, the respective image tag will be overwritten with an empty image.\n", ref.Context().RegistryStr())
+
+			// In case the input argument included a digest, the reference
+			// needs to be updated to exclude the digest for the empty image
+			// override to succeed.
+			switch ref.(type) {
+			case name.Digest:
+				tmp := strings.SplitN(image, "@", 2)
+				ref, err = name.NewTag(tmp[0])
+				if err != nil {
+					return err
+				}
+			}
+
+			return remote.Write(
+				ref,
+				empty.Image,
+				remote.WithContext(ctx),
+				remote.WithAuth(auth),
+			)
+		}
+
+	default:
+		return remote.Delete(
+			ref,
+			remote.WithContext(ctx),
+			remote.WithAuth(auth),
+		)
+	}
+}
+
+func dockerHubLogin(username string, password string) (string, error) {
+	type LoginData struct {
+		Username string `json:"username"`
+		Password string `json:"password"`
+	}
+
+	loginData, err := json.Marshal(LoginData{Username: username, Password: password})
+	if err != nil {
+		return "", err
+	}
+
+	req, err := http.NewRequest("POST", "https://hub.docker.com/v2/users/login/", bytes.NewReader(loginData))
+	if err != nil {
+		return "", err
+	}
+
+	req.Header.Set("Content-Type", "application/json")
```

Comment:
Given that you JSON parse the response.

```suggestion
	req.Header.Set("Accept", "application/json")
	req.Header.Set("Content-Type", "application/json")
```